workspace/manager.py

```
@@ -240,12 +253,157 @@ def create_workspace(
+        """
+        Perform a shallow, non-checkout clone to validate repository access for hosts that do not support a Contents API.
+        
+        This is a heavier access check than `ls-remote`; it always removes any temporary files it creates and should be used only when lighter checks are insufficient.
+        
+        Returns:
+            result (dict[str, object]): A dictionary with `ok` (`True` if access validated, `False` otherwise) and `error` (an error message string or `None`).
+        """
+        try:
+            from workspace.dry_clone import dry_clone_repo
+            return dry_clone_repo(repo_url, token, timeout)
+        except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)